### PR TITLE
Mon 59116 previous otel pr comments

### PR DESCRIPTION
**Pull request by jean-christophe81**
## Description

Some class has been renamaed

Please include a short resume of the changes and what is the purpose of PR. Any relevant information should be added to help:
* **QA Team** (Quality Assurance) with tests.
* **reviewers** to understand what are the stakes of the pull request.

**Fixes** # (issue)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [ ] master

<h2> How this pull request can be tested ? </h2>

Please describe the **procedure** to verify that the goal of the PR is matched. Provide clear instructions so that it can be **correctly tested**.

Any **relevant details** of the configuration to perform the test should be added.

## Checklist

- [ ] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [ ] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: tests/resources/opentelemetry/telegraf2.conf [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+[agent]
+  ## Default data collection interval for all inputs
+  interval = "60s"
+
+[[outputs.opentelemetry]]
+   service_address = "127.0.0.1:4317"
+
+
+[[inputs.exec]]
+  name_override = "host_3__service_5"
```
Comment (Evan-Adam) [inline tests/resources/opentelemetry/telegraf2.conf:12]: Noob question here, is it normal that you have two underscode in the name override ?
should'nt it be host_3_service_5 ?

**Review by Evan-Adam — APPROVED** [review]

File: tests/resources/opentelemetry/telegraf2.conf (hunk 1 above)
Comment (jean-christophe81) [reply tests/resources/opentelemetry/telegraf2.conf]: I have forgotten to remove these files

Comment (jean-christophe81) [reply tests/resources/opentelemetry/telegraf2.conf]: removed

File: engine/inc/com/centreon/engine/commands/otel_interface.hh [hunk 2]
```diff
@@ -67,17 +67,17 @@ class host_serv_list {
   void unregister_host_serv(const std::string& host,
                             const std::string& service_description);
```
Comment (bouda1) [inline engine/inc/com/centreon/engine/commands/otel_interface.hh:68]: why not `remove` method?

File: engine/inc/com/centreon/engine/commands/otel_interface.hh [hunk 3]
```diff
@@ -67,17 +67,17 @@ class host_serv_list {
...
-  host_serv_metric is_allowed(const host_set& hosts,
-                              const service_set& services) const;
+  host_serv_metric contains(const host_set& hosts,
+                            const service_set& services) const;
 };
 
 template <typename host_set, typename service_set>
-host_serv_metric host_serv_list::is_allowed(const host_set& hosts,
-                                            const service_set& services) const {
+host_serv_metric host_serv_list::contains(const host_set& hosts,
```
Comment (bouda1) [inline engine/inc/com/centreon/engine/commands/otel_interface.hh:79]: contains is not good here.
My choice would be 'match'

It remebers me api for regexp. And it is their function for that.

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh [hunk 4]
```diff
@@ -20,6 +20,18 @@
 #define CCE_MOD_OTL_NAGIOS_CONVERTER_HH
 
 namespace com::centreon::engine::modules::opentelemetry::telegraf {
+
+/**
+ * @brief nagios converter doesn't accept config parameter so it is only used
+ * for the check_result_builder_config::converter_type::nagios_converter
+ */
+class nagios_converter_config : public check_result_builder_config {
```
Comment (bouda1) [inline engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh:28]: Why not continuing the change also here?

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh [hunk 5]
```diff
@@ -91,7 +103,7 @@ namespace com::centreon::engine::modules::opentelemetry::telegraf {
  *
  *
  */
-class nagios_converter : public otl_converter {
+class nagios_converter : public otl_check_result_builder {
```
Comment (bouda1) [inline engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh:106]: and here.
nagios_check_result_builder

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh (hunk 4 above)
Comment (bouda1) [reply engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh:28]: nagios_check_result_builder_config

File: engine/inc/com/centreon/engine/commands/otel_interface.hh (hunk 3 above)
Comment (jean-christophe81) [reply engine/inc/com/centreon/engine/commands/otel_interface.hh:79]: done

File: engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh (hunk 4 above)
Comment (jean-christophe81) [reply engine/modules/opentelemetry/inc/com/centreon/engine/modules/opentelemetry/telegraf/nagios_converter.hh:28]: pb rebase

**Review by bouda1 — APPROVED** [review]

**Pull request merged**